osu.Game/Overlays/Mods/ModButton.cs

```diff
@@ -64,41 +64,45 @@ private bool changeSelectedIndex(int newIndex)
             if (newIndex >= 0 && !Mods[newIndex].HasImplementation)
                 return false;
 
-            selectedIndex = newIndex;
-            Mod modAfter = SelectedMod ?? Mods[0];
-
-            if (beforeSelected != Selected)
+            Schedule(() =>
```

Comment:
Think it might be worth to split the body of the schedule to a method, call it `runModSelectionTransforms()` or similar?

Reply:
It shares four variables from above. could make it a local function but not sure it improves much.

Reply:
On revisiting, doesn't seem to split well (4 args). Let's just go with this as written.